File: VB/SchedulerSource.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CustomSorting
	Public Class SchedulerData
		Private Const daysInWeek As Integer = 7
		Private employees() As String = { "Jane", "Martin", "John", "Jack", "Amanda", "Carmen", "Wins", "Todd", "Ashley" }
		Private rnd As New Random()
		Public Property Items() As List(Of SchedulerItem)
		Public Sub New()
			Items = New List(Of SchedulerItem)()
			GenerateRandomData()
		End Sub
		Private Sub GenerateRandomData()
			For i As Integer = 0 To daysInWeek - 1
				Dim e1 As Integer = rnd.Next(employees.Length - 1)
				Dim e2 As Integer = e1 + 1
				Dim day As String = Date.Today.AddDays(i).DayOfWeek.ToString()
				Items.Add(New SchedulerItem() With {.Day = day, .Employee = employees(e1)})
				Items.Add(New SchedulerItem() With {.Day = day, .Employee = employees(e2)})
			Next i
		End Sub
	End Class

	Public Class SchedulerItem
		Public Property Day() As String
		Public Property Employee() As String
	End Class
End Namespace

File: CS/SchedulerSource.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomSorting {
    public class SchedulerData {
        const int daysInWeek = 7;
        string[] employees = new string[] { "Jane", "Martin", "John", "Jack", "Amanda", "Carmen", "Wins", "Todd", "Ashley" };
        Random rnd = new Random();
        public List<SchedulerItem> Items { get; set; }
        public SchedulerData() {
            Items = new List<SchedulerItem>();
            GenerateRandomData();
        }
        void GenerateRandomData() {
            for (int i = 0; i < daysInWeek; i++) {
                int e1 = rnd.Next(employees.Length - 1);
                int e2 = e1 + 1;
                string day = DateTime.Today.AddDays(i).DayOfWeek.ToString();
                Items.Add(new SchedulerItem() { Day = day, Employee = employees[e1] });
                Items.Add(new SchedulerItem() { Day = day, Employee = employees[e2] });
            }
        }
    }

    public class SchedulerItem {
        public string Day { get; set; }
        public string Employee { get; set; }
    }
}
